app/src/main/java/io/apicurio/registry/rules/RulesServiceImpl.java

```diff
@@ -61,17 +66,26 @@ public void applyRules(String artifactId, ArtifactType artifactType, ContentHand
             rules = storage.getGlobalRules();
             useGlobalRules = true;
         }
-        if (rules.isEmpty()) {
-            return;
+
+        List<RuleType> defaultGlobalRules = rulesProperties.getDefaultGlobalRules(rules);
+        if (!defaultGlobalRules.isEmpty()) {
+            // We have default global rules as well as rules from storage, so use the concatenated list
+            rules = Stream.concat(rules.stream(), defaultGlobalRules.stream())
+                .collect(Collectors.toList());

```

Comment:
This could be me misunderstanding how `concat()` works, but won't this be a problem if the same rule exists in both?  We don't want to run the rules twice.  But maybe `concat` de-dupes the items in the stream?

Reply:
This is down to my badly-named method again. Hopefully it's clearer now - in this case `defaultGlobalRules` only contains configured rules that are _not_ found in the rules from storage.